resources/fc_cleanup/cleanup.py

```diff
@@ -0,0 +1,100 @@
+"""Cleanup duplicate File Catalog records...
+
+where the duplicate is indexed under the file's realpath.
+"""
+
+
+import argparse
+import json
+import logging
+import os
+from typing import Any, cast, Dict, Generator, Tuple
+
+import coloredlogs  # type: ignore[import]
+
+# local imports
+from rest_tools.client import RestClient  # type: ignore[import]
+
+BATCH_SIZE = 10000
+
+FCFile = Dict[str, Any]
+
+
+def _get_fcfile(rc: RestClient, logical_name: str) -> FCFile:
+    body = {
+        "query": json.dumps({"logical_name": logical_name}),
+        "keys": "logical_name|checksum",
+    }
+
+    try:
+        results = rc.request_seq("GET", "/api/files", body)["files"]
+    except KeyError:
+        raise FileNotFoundError
+    if not results:
+        raise FileNotFoundError
+    if len(results) > 1:
+        raise Exception(f"Multiple FC matches for {logical_name}")
+
+    return cast(FCFile, results[0])
+
+
+def get_evil_twin(rc: RestClient, this: FCFile) -> FCFile:
+    """Get the FC file that is indexed under `this`'s realpath."""
+    realpath = os.path.realpath(this["logical_name"])
+    evil_twin = _get_fcfile(rc, realpath)
+
+    # TODO - what other fields to check?
+    if this["checksum"]["sha512"] != evil_twin["checksum"]["sha512"]:
+        raise Exception(
+            f"Checksums don't match {this['logical_name']} vs {evil_twin['logical_name']}"
+        )
```

Comment:
Would probably want to verify that all filled in fields in `evil_twin` are also in `this`.

Reply:
Okay, how about this now?

Reply:
Okay, I've flipped this logic now. I will also need to do something about the locations' path value since that is essentially the logical_name as well

Reply:
> Okay, I've flipped this logic now. I will also need to do something about the locations' path value since that is essentially the logical_name as well

I'm ignoring that value now as well.